modules/theme/plugin.js

```diff
@@ -0,0 +1,44 @@
+import Vue from 'vue'
+import { parse, serialize } from 'cookie'
+
+const key = '<%= options.cookie.key %>'
+const cookieOptions = <%= JSON.stringify(options.cookie.options, null, 2) %>
+
+export default function (ctx, inject) {
+  const theme = Vue.observable({
+    value: (process.client ? window.__nuxt_theme : '<%= options.value %>')
+  })
+  const setTheme = (value, persist = true) => {
+    theme.value = value
+    if (process.client) {
```

Comment:
Maybe a todo for future improvements. One may call `$theme` on the server-side which we will lose it in this case. We can use `res` to set cookie :)

Reply:
Future improvement yes 👍